# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditorDocumentCommands.cpp
# Commit: 9ee7acab98a4
# Full Hash: 9ee7acab98a470b2860125ba3deb439a64dc83e7
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-06-10 21:47:35
# Regressor Bug: 1529884
# File Overlap Count: 1
# Description:
#   Bug 1529884 - part 6: Through subject principal at Document::ExecCommand() to constructor of EditorBase::AutoEditActionDataSetter r=smaug
#   
#   `Document::ExecCommand()` knows subject principal.  This patch makes it tell
#   `EditorCommand::DoCommand()` and `EditorCommand::DoCommandParam()`.  Then,
#   makes they tell each editor public methods which may cause dispatching
# ==============================================================================

diff -r 258c4da663ca -r 9ee7acab98a4 editor/libeditor/HTMLEditorDocumentCommands.cpp
--- a/editor/libeditor/HTMLEditorDocumentCommands.cpp	Mon Jun 10 10:26:04 2019 +0000
+++ b/editor/libeditor/HTMLEditorDocumentCommands.cpp	Mon Jun 10 10:27:07 2019 +0000
@@ -11,6 +11,7 @@
 #include "nsCommandParams.h"         // for nsCommandParams
 #include "nsIDocShell.h"             // for nsIDocShell
 #include "nsIEditingSession.h"       // for nsIEditingSession, etc
+#include "nsIPrincipal.h"            // for nsIPrincipal
 #include "nsISelectionController.h"  // for nsISelectionController
 #include "nsISupportsImpl.h"         // for nsPresContext::Release
 #include "nsISupportsUtils.h"        // for NS_IF_ADDREF
@@ -44,13 +45,14 @@
 }
 
 nsresult SetDocumentStateCommand::DoCommand(Command aCommand,
-                                            TextEditor& aTextEditor) const {
+                                            TextEditor& aTextEditor,
+                                            nsIPrincipal* aPrincipal) const {
   return NS_ERROR_NOT_IMPLEMENTED;
 }
 
 nsresult SetDocumentStateCommand::DoCommandParam(
-    Command aCommand, const Maybe<bool>& aBoolParam,
-    TextEditor& aTextEditor) const {
+    Command aCommand, const Maybe<bool>& aBoolParam, TextEditor& aTextEditor,
+    nsIPrincipal* aPrincipal) const {
   if (NS_WARN_IF(aBoolParam.isNothing())) {
     return NS_ERROR_INVALID_ARG;
   }
@@ -123,8 +125,8 @@
 }
 
 nsresult SetDocumentStateCommand::DoCommandParam(
-    Command aCommand, const nsACString& aCStringParam,
-    TextEditor& aTextEditor) const {
+    Command aCommand, const nsACString& aCStringParam, TextEditor& aTextEditor,
+    nsIPrincipal* aPrincipal) const {
   if (NS_WARN_IF(aCStringParam.IsVoid())) {
     return NS_ERROR_INVALID_ARG;
   }
@@ -356,7 +358,8 @@
 }
 
 nsresult DocumentStateCommand::DoCommand(Command aCommand,
-                                         TextEditor& aTextEditor) const {
+                                         TextEditor& aTextEditor,
+                                         nsIPrincipal* aPrincipal) const {
   return NS_ERROR_NOT_IMPLEMENTED;
 }
 